includes/templates/template_default/templates/tpl_modules_shipping_estimator.php

```diff
@@ -3,15 +3,15 @@
  * Module Template - for shipping-estimator display
  *
  * @package templateSystem
- * @copyright Copyright 2003-2019 Zen Cart Development Team
+ * @copyright Copyright 2003-2019 Zen Cart Development Team
  * @copyright Portions Copyright 2003 osCommerce
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id: mc12345678 2019 Apr 30 Modified in v1.5.6b $
+ * @version $Id: mc12345678 2019 Apr 30 Modified in v1.5.6b $
  */
 ?>
 <div id="shippingEstimatorContent">
 <?php echo zen_draw_form('estimator', zen_href_link($show_in . '#view', '', $request_type), 'post'); ?>
-<?php echo zen_draw_hidden_field('scid', $selected_shipping['id']); ?>
+<?php echo zen_draw_hidden_field('scid', is_array($selected_shipping) ? $selected_shipping['id'] : ''); ?>

```

Comment:
I think I wouldn't write the hidden field unless the data was there - i.e. 
```
if (is_array($selected_shipping)) {
   echo zen_draw_hidden_field('scid', $selected_shipping['id']);
}
```

Reply:
Doh, still baby steps.